From uli+cip@fpond.eu  Sun Dec  4 16:59:13 2022
From: Ulrich Hecht <uli+cip@fpond.eu>
Date: Fri,  2 Dec 2022 05:42:53 +0100
Subject: Revert "fbdev: fb_pm2fb: Avoid potential divide by zero error"
To: stable@vger.kernel.org
Cc: sashal@kernel.org, gregkh@linuxfoundation.org, Ulrich Hecht <uli+cip@fpond.eu>
Message-ID: <20221202044253.516827-1-uli+cip@fpond.eu>

From: Ulrich Hecht <uli+cip@fpond.eu>

This reverts commit 6577e903a9e193ad70f2db92eba57c4f335afd1a. It's a
duplicate of a commit that is already in this tree
(0f1174f4972ea9fad6becf8881d71adca8e9ca91).

Signed-off-by: Ulrich Hecht <uli+cip@fpond.eu>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/video/fbdev/pm2fb.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/video/fbdev/pm2fb.c
+++ b/drivers/video/fbdev/pm2fb.c
@@ -619,11 +619,6 @@ static int pm2fb_check_var(struct fb_var
 		return -EINVAL;
 	}
 
-	if (!var->pixclock) {
-		DPRINTK("pixclock is zero\n");
-		return -EINVAL;
-	}
-
 	if (PICOS2KHZ(var->pixclock) > PM2_MAX_PIXCLOCK) {
 		DPRINTK("pixclock too high (%ldKHz)\n",
 			PICOS2KHZ(var->pixclock));
